src/OrchardCore/OrchardCore.Setup.Abstractions/Events/ISetupEventHandler.cs

```diff
@@ -2,24 +2,29 @@
 
 namespace OrchardCore.Setup.Events;
 
+/// <summary>
+/// Defines methods for handling setup events, including performing setup operations and marking their success or
+/// failure.
+/// </summary>
 public interface ISetupEventHandler
 {
     /// <summary>
-    /// Called during the process of setting up a new tenant.
+    /// Performs asynchronous setup operations using the specified context.
     /// </summary>
-    /// <param name="context"></param>
-    /// <returns></returns>
+    /// <param name="context">An object that provides configuration and state information required for the setup process. Cannot be null.</param>
+    /// <returns>A task that represents the asynchronous setup operation.</returns>
     Task SetupAsync(SetupContext context) => Task.CompletedTask;
 
     /// <summary>
-    /// Called when a tenant fails to setup.
+    /// Marks the setup operation as failed within the provided context.
```

Comment:
This is wrong. It's also a notification only, it does not mark the operation as failed.